lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/statemachine/TransitionResult.java

```diff
@@ -12,6 +14,40 @@ public record TransitionResult(
         Map<String, String> auditContext,
         String targetEntryEvent) {
     public TransitionResult(State state) {
-        this(state, null, null, null);
+        this(state, new ArrayList<>(), new HashMap<>(), null);
+    }
+
+    public static List<AuditEventTypes> mergeAuditEvents(
```

Comment:
Do you think it's worth making these generic collection utils - i.e. to merge nullable collections together? I don't think there's anything specific about audit events in them.

Reply:
I'll do that - I was initially concerned that the behaviour of returning an empty collection when both inputs are null was maybe not generic enough. But on reflection it seems like a good idea.